Examples/AsianOption/AsianOption.cpp

```diff
@@ -0,0 +1,314 @@
+
+        // Monte Carlo arithmetic discrete with control variate
+        std::cout << std::setw(widths[0]) << std::left << "MC with Control Variate"
+                  << std::fixed << std::setprecision(6)
+                  << std::setw(widths[1]) << std::left;
+        discreteArithmeticOption.setPricingEngine(
+            MakeMCDiscreteArithmeticAPEngine<PseudoRandom>(bsmProcess)
+                .withSamples(10000)
+                .withControlVariate()
+                .withSeed(mcSeed));
+        std::cout << discreteArithmeticOption.NPV() << std::endl;
+
+        // Turnbull-Wakeman approximation
+        std::cout << std::setw(widths[0]) << std::left << "Turnbull-Wakeman Approximation"
+                  << std::fixed << std::setprecision(6)
+                  << std::setw(widths[1]) << std::left;
+        discreteArithmeticOption.setPricingEngine(
+            ext::make_shared<TurnbullWakemanAsianEngine>(bsmProcess));
+        std::cout << discreteArithmeticOption.NPV() << std::endl;
+
+        // **********************************************
+        // CONTINUOUS AVERAGING - GEOMETRIC AVERAGE
+        // **********************************************
+        std::cout << std::endl;
+        std::cout << "CONTINUOUS AVERAGING - GEOMETRIC AVERAGE" << std::endl;
+        std::cout << "========================================" << std::endl;
+
+        // Analytic geometric continuous
+        std::cout << std::setw(widths[0]) << std::left << "Analytic (Continuous Geometric)"
+                  << std::fixed << std::setprecision(6)
+                  << std::setw(widths[1]) << std::left;
+        continuousGeometricOption.setPricingEngine(
+            ext::make_shared<AnalyticContinuousGeometricAveragePriceAsianEngine>(bsmProcess));
+        std::cout << continuousGeometricOption.NPV() << std::endl;
+
+        // **********************************************
+        // CONTINUOUS AVERAGING - ARITHMETIC AVERAGE
+        // **********************************************
+        std::cout << std::endl;
+        std::cout << "CONTINUOUS AVERAGING - ARITHMETIC AVERAGE" << std::endl;
+        std::cout << "=========================================" << std::endl;
+
+        // Continuous Arithmetic Levy Engine
+        std::cout << std::setw(widths[0]) << std::left << "Continuous Arithmetic Levy Engine"
+                  << std::fixed << std::setprecision(6)
+                  << std::setw(widths[1]) << std::left;
+        auto currentAverage = makeQuoteHandle(0.0);  // No averaging yet for fresh option
+        continuousArithmeticOption.setPricingEngine(
+            ext::make_shared<ContinuousArithmeticAsianLevyEngine>(bsmProcess, currentAverage, settlementDate));
+        std::cout << continuousArithmeticOption.NPV() << std::endl;
+
+        // **********************************************
+        // PDE APPROACH (Finite Differences)
+        // **********************************************
```

Comment:
It seems odd to have this singled out.  I would have expected this to go together with the other methods for discrete arithmetic averaging (MC, Turnbull...)